types/socket.io/socket.io-tests.ts

```diff
@@ -235,3 +235,16 @@ function testServerEventEmitter() {
     io.setMaxListeners(50);
     io.rawListeners('event');
 }
+
+function testOverwriteGenerateId() {
+    var io = socketIO.listen(80);
+    io.use((socket, next) => {
+        io.engine.generateId = () => {
+            return socket.handshake.query.deviceCode;
+        }
+        next();
+    })
+    .on('connection', (socket) => {
+        console.log(socket.id);
```

Comment:
please add something to throw if it is not as expected.

Reply:
Ah okay

Reply:
Is this better?

Reply:
Waiting for the results of the test.